src/inc/til/string.h

```diff
@@ -59,15 +61,110 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
 #pragma warning(disable : 26481) // Don't use pointer arithmetic. Use span instead (bounds.1).
         return str.size() >= prefix.size() && Traits::compare(str.data() + (str.size() - prefix.size()), prefix.data(), prefix.size()) == 0;
 #pragma warning(pop)
-    };
+    }
 
-    constexpr bool ends_with(const std::string_view str, const std::string_view prefix) noexcept
+    constexpr bool ends_with(const std::string_view& str, const std::string_view& prefix) noexcept
     {
         return ends_with<>(str, prefix);
-    };
+    }
 
-    constexpr bool ends_with(const std::wstring_view str, const std::wstring_view prefix) noexcept
+    constexpr bool ends_with(const std::wstring_view& str, const std::wstring_view& prefix) noexcept
     {
         return ends_with<>(str, prefix);
-    };
+    }
+
+    // Just like std::tolower, but without annoying locales.
+    template<typename T>
+    constexpr T tolower_ascii(T c)
+    {
+        if ((c >= 'A') && (c <= 'Z'))
+        {
+            c |= 0x20;
+        }
+
+        return c;
+    }
+
+    // Just like std::toupper, but without annoying locales.
+    template<typename T>
+    constexpr T toupper_ascii(T c)
+    {
+        if ((c >= 'a') && (c <= 'z'))
+        {
+            c &= ~0x20;
+        }
+
+        return c;
+    }
+
+    // Just like _memicmp, but without annoying locales.
+    template<typename T, typename Traits>
+    _TIL_INLINEPREFIX bool equals_insensitive_ascii(const std::basic_string_view<T, Traits>& str1, const std::basic_string_view<T, Traits>& str2) noexcept

```

Comment:
I can write some unit tests for those. I know they work, because I wrote tests in my `test.sln` project, but I didn't add them to this PR yet. 😅